src/main/java/at/hannibal2/skyhanni/features/event/diana/ArrowGuessBurrow.kt

```diff
@@ -0,0 +1,223 @@
+package at.hannibal2.skyhanni.features.event.diana
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.ReceiveParticleEvent
+import at.hannibal2.skyhanni.events.diana.BurrowGuessEvent
+import at.hannibal2.skyhanni.features.event.diana.GriffinBurrowHelper.allowedBlocksAboveGround
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.BlockUtils.getBlockAt
+import at.hannibal2.skyhanni.utils.BlockUtils.isInLoadedChunk
+import at.hannibal2.skyhanni.utils.LocationUtils.isInside
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.RaycastUtils
+import at.hannibal2.skyhanni.utils.collection.TimeLimitedSet
+import net.minecraft.init.Blocks
+import net.minecraft.util.EnumParticleTypes
+import kotlin.time.Duration.Companion.minutes
+
+@SkyHanniModule
+object ArrowGuessBurrow {
+    private val config get() = SkyHanniMod.feature.event.diana
+
+    private val points: HashSet<LorenzVec> = HashSet()
+    private val recentArrowParticles = TimeLimitedSet<LorenzVec>(1.minutes)
+
+    @HandleEvent(onlyOnIsland = IslandType.HUB, receiveCancelled = true)
+    fun onReceiveParticle(event: ReceiveParticleEvent) {
+        if (!isEnabled()) return
+
+        if (event.distanceToPlayer > 6) return
+        if (event.type != EnumParticleTypes.REDSTONE) return
+        if (event.count != 0) return
+        if (event.speed != 1.0f) return
+
+        val allowedOffsets = setOf(0.0, 128.0, 255.0)
+        if (event.offset.x !in allowedOffsets) return
+        if (event.offset.y !in allowedOffsets) return
+        if (event.offset.z !in allowedOffsets) return
```

Comment:
you can replace this with
```suggestion
        if (!event.offset.toDoubleArray().all(allowedOffsets::contains)) return
```